core/cloudflow-cli/src/main/scala/akka/cli/cloudflow/kubeclient/KubeClientFabric8.scala

```diff
@@ -50,22 +52,25 @@ object KubeClientFabric8 {
 }
 
 class KubeClientFabric8(
-    val config: Option[File],
-    clientFactory: Config => KubernetesClient = new DefaultKubernetesClient(_))(implicit val logger: CliLogger)
+    val configFile: Option[File],
+    clientFactory: K8sConfig => KubernetesClient = new DefaultKubernetesClient(_))(implicit val logger: CliLogger)
     extends KubeClient {
   import KubeClientFabric8._
 
   private lazy val kubeClient = Try {
-    def getConfig() = {
```

Comment:
Not doing a full review now but this is intentionally a `def`, to workaround a fabric8 issue.

In general, although I like some of the stylistic improvements you are doing, do you mind to separate in 2 PR? One only with the changes strictly needed for the feature and one with the additional improvements. This will make it easier to review :+1: